orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/gce/ModifyGceAutoscalingPolicyStage.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.pipeline.providers.gce;
+
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroup;
+import com.netflix.spinnaker.orca.clouddriver.pipeline.servergroup.support.TargetServerGroupLinearStageSupport;
+import com.netflix.spinnaker.orca.clouddriver.tasks.MonitorKatoTask;
+import com.netflix.spinnaker.orca.clouddriver.tasks.providers.gce.autoscaling.UpsertGceAutoscalingPolicyTask;
+import com.netflix.spinnaker.orca.clouddriver.tasks.servergroup.ServerGroupCacheForceRefreshTask;
+import com.netflix.spinnaker.orca.clouddriver.utils.OortHelper;
+import com.netflix.spinnaker.orca.pipeline.TaskNode;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+@Slf4j
+@Component
+public class ModifyGceAutoscalingPolicyStage extends TargetServerGroupLinearStageSupport {
+
+  @Override
+  protected void taskGraphInternal(Stage stage, TaskNode.Builder builder) {
+    builder
+      .withTask("upsert", UpsertGceAutoscalingPolicyTask.class)
+      .withTask("monitor", MonitorKatoTask.class)
+      .withTask("forceCacheRefresh", ServerGroupCacheForceRefreshTask.class)
+      .withTask("waitForScalingProcesses", WaitForAutoscalingPolicy.class);
+  }
+
+  @Component
+  public static class WaitForAutoscalingPolicy implements RetryableTask {
```

Comment:
Should this task also have `Gce` in its name?

Reply:
Any particular reason to embed this class?

Reply:
Sure, split it into a separate `Task` class and added 'Gce' and appended 'Task'.